pkg/cloudprovider/providers/gce/gce_zones.go

```diff
@@ -43,14 +43,34 @@ func (gce *GCECloud) GetZone() (cloudprovider.Zone, error) {
 // This is particularly useful in external cloud providers where the kubelet
 // does not initialize node data.
 func (gce *GCECloud) GetZoneByProviderID(providerID string) (cloudprovider.Zone, error) {
-	return cloudprovider.Zone{}, errors.New("GetZoneByProviderID not implemented")
+	_, zone, _, err := splitProviderID(providerID)
+	if err != nil {
+		return cloudprovider.Zone{}, err
+	}
+	region, err := GetGCERegion(zone)
+	if err != nil {
+		return cloudprovider.Zone{}, err
+	}
+	return cloudprovider.Zone{FailureDomain: zone, Region: region}, nil
 }
 
 // GetZoneByNodeName implements Zones.GetZoneByNodeName
 // This is particularly useful in external cloud providers where the kubelet
 // does not initialize node data.
 func (gce *GCECloud) GetZoneByNodeName(nodeName types.NodeName) (cloudprovider.Zone, error) {
-	return cloudprovider.Zone{}, errors.New("GetZoneByNodeName not imeplemented")
+	list, err := gce.service.Zones.List(gce.projectID).Filter(fmt.Sprintf("name eq %s", mapNodeNameToInstanceName(nodeName))).Do()
```

Comment:
I'm not convinced about this. According to this documentation:
https://godoc.org/google.golang.org/api/compute/v1#ZonesService.List

It seems that Zones.List is listing zones.
So how passing instanceName as a filter for the name is supposed to work here?